scipy/signal/fir_filter_design.py

```diff
@@ -791,3 +799,200 @@ def firls(numtaps, bands, desired, weight=None, nyq=1.):
     # make coefficients symmetric (linear phase)
     coeffs = np.hstack((a[:0:-1], 2 * a[0], a[1:]))
     return coeffs
+
+
+def _dhtm(mag):
+    """Compute the modified 1D discrete Hilbert transform
+
+    Parameters
+    ----------
+    mag : ndarray
+        The magnitude spectrum. Should be an even number, and
+        preferably fast for FFT/IFFT.
+    """
+    # Adapted based on code by Niranjan Damera-Venkata,
+    # Brian L. Evans and Shawn R. McCaslin
```

Comment:
Is there a package name / paper / link you could add for this reference and add that reference to the docstring?